crates/rattler_conda_types/src/repo_data/topological_sort.rs

```diff
@@ -0,0 +1,1623 @@
+use crate::PackageRecord;
+use fxhash::{FxHashMap, FxHashSet};
+
+/// Sorts the packages topologically
+///
+/// This function is deterministic, meaning that it will return the same result regardless of the
+/// order of `packages` and of the `depends` vector inside the records
+pub fn sort_topologically<T: AsRef<PackageRecord>>(packages: Vec<T>) -> Vec<T> {
+    let roots = get_graph_roots(&packages);
+
+    let mut all_packages = packages
+        .into_iter()
+        .map(|p| (p.as_ref().name.clone(), p))
+        .collect();
+
+    get_topological_order(roots, &mut all_packages)
+}
+
+/// Retrieves the names of the packages that form the roots of the graph
+fn get_graph_roots<T: AsRef<PackageRecord>>(records: &[T]) -> Vec<String> {
```

Comment:
I think this function might fall apart if we have multiple package records with the same name in the vector, right?

Since version 0 might be at the root, but if version 1 adds some dependencies it might not be at the root anymore. 

We might want to clarify that this works only for lists of unique package names.

Reply:
Good one, I added this to the docs.